pkgs/development/haskell-modules/lib.nix

```diff
@@ -126,6 +126,10 @@ rec {
    */
   dontDistribute = drv: overrideCabal drv (drv: { hydraPlatforms = []; });
 
+  /* overridePlatforms changes the supported platforms for a package.
+   */
+  overridePlatforms = drv: fn: overrideCabal drv (drv: { platforms = fn pkgs.lib.platforms; });
```

Comment:
Not sure if I'm too happy with using a function here — there is really no need to as there is only one `platforms` set (as opposed to other cases where we use this type of API like `ghcWithPackages` where there are different sets the function could take packages from).

I guess the shortcut is nice, but I fear we introduce a unnecessary and potentially unintuitive API here. Interested to hear other opinions.

Reply:
I don't disagree, and it was definitely for convenience only. My original plan was to see if I could accomplish this with even less boilerplate, as maybe `platformOverrides = { honk = platforms.linux; ... }`, but I went for something like this to keep the diff small this close to release.

So, we can of course inline it all:

```nix
{
  honk = overridePlatforms super.honk pkgs.lib.platforms.linux;
}
```

Or maybe introduce a variable?

```nix
let
  inherit (pkgs.lib) platforms;
in {
  honk = overridePlatforms super.honk platforms.linux;
}
```

Though this file does not contain any yet, and I'm not sure if it's desirable.

Reply:
I've pushed a version that inlines everything.

I also rebased on the merge-base of `haskell-updates` and `master`. Is that sufficient for testing, or do we need to be on the latest `haskell-updates`?

The reason I'm targeting `master` is because of ZHF, and the next `haskell-updates` merge is scheduled post-branchoff: https://github.com/NixOS/nixpkgs/pull/122719

Reply:
No, the next merge is not supposed to be happening past branch-off. The date given there is the usual time window of two weeks.

Reply:
Alright, in that case, I switched target branch and rebased on current `haskell-updates`. 👍